module/VuFind/tests/unit-tests/src/VuFindTest/Hierarchy/TreeDataFormatter/JsonTest.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+/**
+ * Hierarchy TreeDataFormatter Json Test Class
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2010.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+
+namespace VuFindTest\Hierarchy\TreeDataFormatter;
+
+use VuFind\Hierarchy\TreeDataFormatter\Json;
+
+/**
+ * Hierarchy TreeDataFormatter Json Test Class
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+class JsonTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * Invokes a private or protected method and returns its return value.
+     *
+     * @param object $obj     Object
+     * @param string $method  Method name
+     * @param mixed  ...$args Arguments passed to method
+     *
+     * @return mixed
+     */
+    private function invokePrivateMethod($obj, $method, ...$args)
```

Comment:
You can mix in VuFind's ReflectionTrait to get this functionality without having to redefine it in your test. See:

https://github.com/vufind-org/vufind/blob/dev/module/VuFind/src/VuFindTest/Feature/ReflectionTrait.php#L56